### ISSUE TITLE
Request for runSVUnit to exit 1 when a test fails

### ISSUE THREAD

**Issue by jesseprusi** [svunit/svunit#16]
This would eliminate the need to grep for pass/fail status while running SVUnit in regressions.

**Comment by tudortimi** [comment]
No tool implements PASS/FAIL like this. Exit status different than 0 always means that an exceptional situation during the run of a tool happened (e.g. out of memory) and says nothing about the result of that job.

**Comment by jesseprusi** [comment]
I was thinking it could be similar to the exit codes for grep:
0 - completed and a match was found
1 - completed and no match was found
2 - exception condition

**Comment by tudortimi** [comment]
Something like this could be implemented in the runSVUnit script and not in the SV code, otherwise it's going to clash with the simulator's error codes.

**Comment by jesseprusi** [comment]
That's exactly what I was thinking.  I grep the log file in my wrapper script to determine the exit code.  I figured I would bring it up here to see if others would like to use the same approach and it could be put into runSVUnit.

**Comment by nosnhojn** [comment]
how about we simplify this as:

0: passing test
!0: everything between failing test and/or exception and/or world coming to an end

I'm no fan of dealing with simulator exit codes so keeping it simple makes this easy.

yay/nay?

**Comment by jesseprusi** [comment]
That sounds good.

**Comment by pestalella** [comment]
Returning a different code when tests pass or fail would make it much easier to integrate the unit testing into the build process and stop the build when unit tests fail.

**Comment by tudortimi** [comment]
I did a 180 on this. I saw that many other testing frameworks (JUnit, pytest, GoogleTest) also signal test failures with their exit codes.

**Comment by GordonMcGregor** [comment]
yup - it is pretty common/ standard to use exit codes.  grepping for results is much more fragile.

**Comment by tudortimi** [comment]
Note to self: this is not the same as #257. We want to check whether any test failed when the simulator command was successful.

The only way I can think of at the moment is to use some file to mark the status. The file would have to be delete when `runSVUnit` is called. The file would be created from SV code and it would contain the result (PASS/FAIL). `runSVUnit` would read this file to determine the test status and react accordingly. To follow the convention from pytest that we have up to now, it would exit with code 1.
